fly/types/bit_stream/bit_stream_reader.h

```diff
@@ -216,13 +216,18 @@ byte_type BitStreamReader::fill(DataType &buffer, byte_type bytes) noexcept
         detail::BitStreamTraits::is_unsigned_integer_v<DataType>,
         "DataType must be an unsigned integer type");
 
-    m_stream.read(
-        reinterpret_cast<std::ios::char_type *>(&buffer),
-        static_cast<std::streamsize>(bytes));
+    byte_type bytesRead = 0;
 
-    buffer = endian_swap<Endian::Big>(buffer);
+    if (m_stream.read(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/trflynn89/libfly/pullRequest?prid=4794728)